src/core/server/saved_objects/import/import_saved_objects.test.ts

```diff
@@ -589,5 +601,94 @@ describe('#importSavedObjectsFromStream', () => {
       const expectedErrors = errors.map(({ type, id }) => expect.objectContaining({ type, id }));
       expect(result).toEqual({ success: false, successCount: 0, errors: expectedErrors });
     });
+
+    test('early return if import data source objects to non-MDS cluster', async () => {
+      const options = setupOptions(false, testDataSourceId, false);
+      const dsObj = createDataSourceObject();
+      const dsExportedObj = createObject(testDataSourceId);
+      const collectedObjects = [dsObj, dsExportedObj];
+
+      const errors = [
+        {
+          type: dsObj.type,
+          id: dsObj.id,
+          title: dsObj.attributes.title,
+          meta: { title: dsObj.attributes.title },
+          error: { type: 'unsupported_type' },
+        },
+        {
+          type: dsExportedObj.type,
+          id: dsExportedObj.id,
+          title: dsExportedObj.attributes.title,
+          meta: { title: dsExportedObj.attributes.title },
+          error: { type: 'unsupported_type' },
+        },
+      ];
+      getMockFn(collectSavedObjects).mockResolvedValue({
+        errors: [],
+        collectedObjects,
+        importIdMap: new Map(), // doesn't matter
+      });
+      const result = await importSavedObjectsFromStream(options);
+      const expectedErrors = errors.map(({ type, id }) => expect.objectContaining({ type, id }));
+      expect(result).toEqual({ success: false, successCount: 0, errors: expectedErrors });
+    });
+
+    test('early return if import mixed non/data source objects to non-MDS cluster', async () => {
+      const options = setupOptions(false, testDataSourceId, false);
+      const dsObj = createDataSourceObject();
+      const dsExportedObj = createObject(testDataSourceId);
+      const nonDsExportedObj = createObject();
+      const collectedObjects = [dsObj, dsExportedObj, nonDsExportedObj];
+
+      const errors = [
+        {
+          type: dsObj.type,
+          id: dsObj.id,
+          title: dsObj.attributes.title,
+          meta: { title: dsObj.attributes.title },
+          error: { type: 'unsupported_type' },
+        },
+        {
+          type: dsExportedObj.type,
+          id: dsExportedObj.id,
+          title: dsExportedObj.attributes.title,
+          meta: { title: dsExportedObj.attributes.title },
+          error: { type: 'unsupported_type' },
+        },
+      ];
+      getMockFn(collectSavedObjects).mockResolvedValue({
+        errors: [],
+        collectedObjects,
+        importIdMap: new Map(), // doesn't matter
```

Comment:
can we use meaningful comment, or remove it if unnecessary?

Reply:
yeah, thanks for the suggestion